src/wrapper/map.rs

```diff
@@ -35,3 +35,41 @@ pub fn map_put(env: NIF_ENV, map: NIF_TERM, key: NIF_TERM, value: NIF_TERM) -> O
     }
     Some(result)
 }
+
+pub struct MapIterator {
```

Comment:
Since this comment, I reimplemented it so that there's no Iterator implementation in wrapper/map.rs. I need to resolve conflicts, though.